### DISCUSSION TITLE
Time resolution of time fields limited to seconds in openapi. Acceptable?

### DISCUSSION THREAD

**Discussion by nicolasfranck** [ugent-library/old-people-service#52] [Ideas]
The implementation of the openapi by [ogen](https://github.com/ogen-go/ogen) limits the time resolution in time fields by not including the milliseconds. This is done internally by implementing the [json.Marshaler](https://pkg.go.dev/encoding/json#Marshaler), so even with `json.Marshal` you would get the same result.

e.g. https://github.com/ugent-library/people-service/blob/main/api/v1/oas_json_gen.go#L963

```
// MarshalJSON implements stdjson.Marshaler.
func (s *Organization) MarshalJSON() ([]byte, error) {
	e := jx.Encoder{}
	s.Encode(&e)
	return e.Bytes(), nil
}
```

which uses another json package (other than the regular `encoding/json`) named https://github.com/go-faster/jx
where you can manually build every part of the json response: https://github.com/ugent-library/people-service/blob/main/api/v1/oas_json_gen.go#L752

```
{
		e.FieldStart("date_created")
		json.EncodeDateTime(e, s.DateCreated)
}
```

Is this acceptable? This makes it hard to sort records when they are inserted rapidly (especially when they imported for the first time)

**Comment by netsensei** [comment]
See: https://stackoverflow.com/questions/49379006/what-is-the-correct-way-to-declare-a-date-in-an-openapi-swagger-file
